soc/arm/atmel_sam0/common/adc_fixup_sam0.h

```diff
@@ -104,6 +104,19 @@
 #else
 #  define ADC_SAM0_BIASREFBUF(n) 0
 #endif
+
+/*
+ * The following MCLK clock configuration fix-up symbols map to the applicable
+ * APB-specific symbols, in order to accommodate different SoC series with the
+ * ADC core connected to different APBs.
+ */
+#if defined(MCLK_APBDMASK_ADC) || defined(MCLK_APBDMASK_ADC0)
+#  define MCLK_ADC (MCLK->APBDMASK.reg)
+#elif defined(MCLK_APBCMASK_ADC0)
+#  define MCLK_ADC (MCLK->APBCMASK.reg)
+#else
+#  error ADC not supported...
```

Comment:
Why can't this be supported in devicetree?

Reply:
At SAM MCU clock connection to ADC module is located in some processor in ABPD registry and some at APBC.
I don't find any usage at devicetree to set / get valid registry.
SAM Drives uses constraints from ASF headers.